src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs

```diff
@@ -197,11 +198,17 @@ public IEnumerable<int> Levels
         /// </summary>
         public bool IsTopLevel { get; set; }
 
+        public string ValueType
+        {
+            get { return _outputType; }
+            set { _outputType = value; RaisePropertyChanged(nameof(ValueType)); }
+        }
+
         #endregion
 
-        public WatchViewModel(Action<string> tagGeometry) : this(null, null, tagGeometry, true) { }
+        public WatchViewModel(Action<string> tagGeometry) : this(null, null, tagGeometry, TypeCode.Empty, true) { }
 
-        public WatchViewModel(string label, string path, Action<string> tagGeometry, bool expanded = false)
+        public WatchViewModel(string label, string path, Action<string> tagGeometry, TypeCode? outputType = TypeCode.Empty, bool expanded = false)
```

Comment:
this is an API break - even the addition of an optional parameter is a binary break, you will need to add a new constructor instead, you can obsolete this one and add a note to use the new one. You may find you cannot / should not make this new parameter optional or ambiguous calls will be possible.

Reply:
can you explain why you choose typeCode instead of Type?

Reply:
@mjkkirschner we added a new overload to the constructor that takes an object instead of string and from the object makes all the right string conversions using the `GetStringFromObject()`. Did not obsolete the current constructor as this seemed safer and you can make the call if it should or shouldn't be obsolete.